pkg/collector/worker/worker.go

```diff
@@ -148,6 +155,25 @@ func (w *Worker) Run() {
 
 		utilizationTracker.CheckStarted()
 
+		if checkStats, found := expvars.CheckStats(check.ID()); found {
+
+			// Calculate start time of previous execution
+			previousFinishDate := checkStats.LastSuccessDate
```

Comment:
You can use UpdateTimestamp instead of computing last finish time.

Reply:
This is what I was looking for. Thanks for pointing it out